openedx/core/djangoapps/discussions/api/config.py

```diff
@@ -0,0 +1,74 @@
+from typing import Iterable, List, Optional
+
+from django.contrib.sites.models import Site
+from django.db.models import Q
+from opaque_keys.edx.keys import CourseKey
+
+# TODO DiscussionProviderConfig
+# TODO DiscussionPluginConfigData
+
+from .data import CourseDiscussionConfigData, DiscussionPluginConfigData
+from ..models import LearningContextDiscussionConfig
+from ...config_model_utils.models import site_from_org
+
+
+def _org_and_site_from_course_key(context_key: CourseKey):
+    org_key = getattr(context_key, 'org', None)
+    site = site_from_org(org_key)
+    return org_key, site
+
+
+def get_course_discussion_config(course_key: CourseKey) -> Optional[CourseDiscussionConfigData]:
+    """
+    Returns the active discussion configuration for the course.
+
+    Args:
+        course_key (CourseKey): Learning context, currently only a course
+
+    Returns:
+        A :class:`CourseDiscussionConfigData` object with the active configuration for this course.
+        Returns `None` if a discussion tool isn't configured for the course yet.
+
+    """
+    try:
+        course_config = LearningContextDiscussionConfig.objects.get(pk=course_key)
+    except LearningContextDiscussionConfig.DoesNotExist:
+        return None
+    return CourseDiscussionConfigData(
+        course_key=course_key,
+        provider=course_config.provider,
+        plugin_configuration=course_config.plugin_configuration,
+        lti_configuration=course_config.lti_configuration,
+        enabled=course_config.enabled,
+    )
+
+
+def update_course_discussion_config(
+    course_key: CourseKey,
+    updated_config: dict
+) -> Optional[CourseDiscussionConfigData]:
+    """
+    Updates the configuration for the specified course.
+
+    Args:
+        course_key (CourseKey): Learning context, currently only a course
+        updated_config (dict): Update configuration to save for specified course
+
+    Returns:
+        A :class:`CourseDiscussionConfigData` object with the active configuration for this course.
+        Returns `None` if a discussion tool isn't configured for the course yet.
+
+    """
+    try:
+        course_config = LearningContextDiscussionConfig.objects.get(pk=course_key)
+    except LearningContextDiscussionConfig.DoesNotExist:
+        raise CourseDiscussionConfigData.DoesNotExist
+    course_config.plugin_configuration = updated_config
```

Comment:
I think I'd rename this function if it only allows updating the `plugin_configuration` portion of the model.  Either that or change the arguments to be a dict with a plugin_configuration in it, or one argument per field on the model that we allow setting.

Reply:
I just removed these; the abstraction wasn't really helpful since we only have one model now.